From f6dfbfc345fa4825f20037c623c51f93fab97e8c Mon Sep 17 00:00:00 2001
From: c4pp4
Date: Tue, 20 Oct 2026 22:28:59 +0200
Subject: [PATCH 1/1] Use the default mechanism to check XDG_CURRENT_DESKTOP

Don't override the desktop environment of GDesktopAppInfo with the first
XDG_CURRENT_DESKTOP entry and don't treat applications shown in GNOME as
shown everywhere. Passing NULL to g_desktop_app_info_get_show_in()
checks all the XDG_CURRENT_DESKTOP entries (e.g. Unity:Unity7).

Signed-off-by: c4pp4
---
 cinnamon-session/csm-autostart-app.c | 5 +----
 1 file changed, 1 insertion(+), 4 deletions(-)

diff --git a/cinnamon-session/csm-autostart-app.c b/cinnamon-session/csm-autostart-app.c
index 0b08f05..0208c6c 100644
--- a/cinnamon-session/csm-autostart-app.c
+++ b/cinnamon-session/csm-autostart-app.c
@@ -139,12 +139,9 @@ is_disabled (CsmApp *app)
 
         /* Check OnlyShowIn/NotShowIn/TryExec */
         current_desktop = csm_util_get_current_desktop ();
-        g_desktop_app_info_set_desktop_env (current_desktop);
         if (current_desktop != NULL &&
             !g_desktop_app_info_get_show_in (G_DESKTOP_APP_INFO (priv->app_info),
-                                             "GNOME") &&
-            !g_desktop_app_info_get_show_in (G_DESKTOP_APP_INFO (priv->app_info),
-                                             current_desktop)) {
+                                             NULL)) {
                         g_debug ("app %s not for %s",
                                  csm_app_peek_id (app), current_desktop);
                 return TRUE;
-- 
2.53.0

